### PR TITLE
8296454: System.console() shouldn't return null in jshell

### PR DESCRIPTION
**Pull request by lahodaj**
Implements `System.console()` for JShell, based on JDK-8295803. It is only supported for (supported) remote agents.

When a snippet calls a Console method in the remote agent, a request to perform the task is sent to the main process, performed, and the outcomes are sent back to the remote agent, and returned from the Console method.

Please also review the CSR for the newly added API:
https://bugs.openjdk.org/browse/JDK-8299680

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8299680](https://bugs.openjdk.org/browse/JDK-8299680) to be approved

### Issues
 * [JDK-8296454](https://bugs.openjdk.org/browse/JDK-8296454): System.console() shouldn't return null in jshell
 * [JDK-8299680](https://bugs.openjdk.org/browse/JDK-8299680): System.console() shouldn't return null in jshell (**CSR**)

### Reviewers
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**) ⚠️ Review applies to [d58e53f1](https://git.openjdk.org/jdk/pull/11695/files/d58e53f1f42fb1b2b7d074a4103286e51c8c6f58)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/11695/head:pull/11695` \
`$ git checkout pull/11695`

Update a local copy of the PR: \
`$ git checkout pull/11695` \
`$ git pull https://git.openjdk.org/jdk.git pull/11695/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11695`

View PR using the GUI difftool: \
`$ git pr show -t 11695`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11695.diff">https://git.openjdk.org/jdk/pull/11695.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/11695#issuecomment-1373385718)

### CODE REVIEW

**Review by vicente-romero-oracle — APPROVED** [review]
looks sensible

File: src/jdk.jshell/share/classes/jdk/jshell/execution/impl/ConsoleImpl.java [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+                    }
+                };
+            } return reader;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public JdkConsole format(String fmt, Object ...args) {
```
Comment (turbanoff) [inline src/jdk.jshell/share/classes/jdk/jshell/execution/impl/ConsoleImpl.java:196]:
```suggestion
        public JdkConsole format(String fmt, Object ... args) {
```

Comment (jonathan-gibbons) [reply src/jdk.jshell/share/classes/jdk/jshell/execution/impl/ConsoleImpl.java]: There is normally no space before the `...`

**Pull request closed without merging**